coreneuron/io/phase2.cpp

```diff
@@ -458,6 +458,7 @@ void Phase2::set_net_send_buffer(Memb_list** ml_list, const std::vector<int>& pn
         if (ml) {  // needs a NetReceiveBuffer
             NetReceiveBuffer_t* nrb =
                 (NetReceiveBuffer_t*) ecalloc_align(1, sizeof(NetReceiveBuffer_t));
+            assert(ml->_net_receive_buffer == nullptr);
```

Comment:
Just a note, ` struct Memb_list` in `coreneuron/mechanism/mechanism.hpp` can default initialise all pointers to `nullptr`.

Reply:
What is the syntax for that? Right now it is explicitly done in coreneuron/mechanism/register_mech.cpp. Is it just to make every field in ```struct Memb_list``` explicit as in
```
    int* nodeindices = nullptr;
    int* _permute = nullptr;
    double* data = nullptr;
    Datum* pdata = nullptr;
...
```

Reply:
yes, like above.

Reply:
Is it the case that with nullptr that one can no longer rely on ```if (foo){...}``` but needs ```if (foo != nullptr){...}```?
So that when one switches from NULL to nullptr that we need to check all the if statements that involve that variable?

Reply:
No, you shouldn't need to change `if(foo)` to `if(foo != nullptr)`. See [ES.87: Don’t add redundant == or != to conditions](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-if). The type of the pointer itself does not change when you choose to initialise it with `nullptr` instead of `NULL` or `0` (or `int* ptr{};`).

Reply:
Made the changes to nullptr